src/Nethermind/Nethermind.Runner/Ethereum/Steps/StartBlockProducerAuRa.cs

```diff
@@ -144,32 +146,14 @@ protected override void BuildProducer()
 
         protected override TxPoolTxSource CreateTxPoolTxSource(ReadOnlyTxProcessingEnv processingEnv, ReadOnlyTxProcessorSource readOnlyTxProcessorSource)
         {
-            Address.TryParse(_auraConfig?.TxPriorityContractAddress, out Address? txPriorityContractAddress);
-            bool usesTxPriorityContract = txPriorityContractAddress != null;
-            
-            if (usesTxPriorityContract)
-            {
-                _txPriorityContract = new TxPriorityContract(_api.AbiEncoder, txPriorityContractAddress, readOnlyTxProcessorSource); 
-            }
+            if (_api.MainBlockProcessor == null) throw new StepDependencyException(nameof(_api.MainBlockProcessor));
             
-            string? auraConfigTxPriorityConfigFilePath = _auraConfig?.TxPriorityConfigFilePath;
-            bool usesTxPriorityLocalData = auraConfigTxPriorityConfigFilePath != null;
-            if (usesTxPriorityLocalData)
-            {
-                _localDataSource = new TxPriorityContract.LocalDataSource(auraConfigTxPriorityConfigFilePath, _api.EthereumJsonSerializer, _api.LogManager);
-            }
+            (_txPriorityContract, _localDataSource) = TxFilterBuilders.CreateTxPrioritySources(_auraConfig, _api, readOnlyTxProcessorSource);
 
-            if (usesTxPriorityContract || usesTxPriorityLocalData)
+            if (_txPriorityContract != null || _localDataSource != null)
             {
-                DictionaryContractDataStore<TxPriorityContract.Destination> minGasPricesContractDataStore = new DictionaryContractDataStore<TxPriorityContract.Destination>(
-                    new TxPriorityContract.DestinationSortedListContractDataStoreCollection(),
-                    _txPriorityContract?.MinGasPrices,
-                    _api.MainBlockProcessor,
-                    _api.LogManager,
-                    _localDataSource?.GetMinGasPricesLocalDataSource());
-
-                _minGasPricesContractDataStore = minGasPricesContractDataStore;
-                _api.DisposeStack.Push(minGasPricesContractDataStore);                
+                _minGasPricesContractDataStore = TxFilterBuilders.CreateMinGasPricesDataStore(_api, _txPriorityContract, _localDataSource, _api.MainBlockProcessor)!;
```

Comment:
can we consider for the future: https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.options.ioptionsmonitor-1?view=dotnet-plat-ext-3.1
IOptionsMonitor and treat contracts as the same type of option source as other config parts
---
in this way we could have Json, Env, CommandLine, Contract, CustomFile as varios option sources
---
the 'DataSource' naming seems to be covering the fact that it is just a dynamic configuration

Reply:
Not sure if it would easily fit solution (we need to construct actual options), will have to research

Reply:
yeah, this is a biggy, not for now - just wanted to hint at it